### DISCUSSION TITLE
Content Warnings

### DISCUSSION THREAD

**Discussion by lo2dev** [stoatchat/discussions#663] [Feature Suggestions]
So I am posting this here to get some thoughts before making an RFC.

The problem: the term NSFW is not broad enough to make sense in other context besides work. Not everything is centered around work and the term is not specific enough to know what to expect to see in the channel. There are also some subjects some people might want to avoid which do not fit into the "mature" category. Examples: channels talking about sensitive topics, if the channel is about mature content it could be specific about what specifically is mature about it

The solution: rename NSFW to Content Warning and add an additional text field in the channel settings to input content warnings as users see fit.

An example of a prompt: 

**Content warning**
This channel includes the following content warnings: talk about suicide and trauma

- [ ] I acknowledge the warning and wish to proceed
(I'm a bit out of ideas to make the acknowledgement sound right for cases where the channel is about porn for example)

**Comment by aisuleep** [comment]
Sounds like a great idea. However, in order to handle for when the channel is for mature content, perhaps the NSFW toggle should not be replaced, but rather an option along with the content warning feature. This way, people will still be prompted for their age to access the mature content, even if someone forgets a content warning.

**Reply by Rexogamer** [reply]
On this note (reposting from the Lounge):
![image](https://github.com/revoltchat/revolt/assets/42586271/bf8be8eb-f4c3-4486-b259-08e3fa332619)

**Reply by aisuleep** [reply]
Oh, perhaps to address the issue of the name being vague, the NSFW toggle can instead be titled, "Mature Content", for more clarification. Other than that, I'm still keeping the idea that it shouldn't be replaced by the more detailed content warning feature; Especially since many people don't bother using content warnings, while the mature toggle is required to be marked when appropriate.

**Reply by lo2dev** [reply]
We could have an additional checkbox saying "This channel is used for age restricted content" while still retaining the custom content warning. Having been on the ferdiverse for a while people will definitely  use content warnings if they have the option.
